subprojects/docs/src/snippets/files/fileCollections/groovy/build.gradle

```diff
@@ -16,26 +16,28 @@ file('src2/dir2').mkdirs()
 
 // tag::closure[]
 tasks.register('list') {
+    Directory projectDirectory = layout.projectDirectory
     doLast {
         File srcDir
 
         // Create a file collection using a closure
-        collection = layout.files { srcDir.listFiles() }
+        collection = projectDirectory.files { srcDir.listFiles() }
 
-        srcDir = file('src')
+        srcDir = projectDirectory.file('src').asFile
         println "Contents of $srcDir.name"
-        collection.collect { relativePath(it) }.sort().each { println it }
+        collection.collect { projectDirectory.asFile.toPath().relativize(it.toPath()) }.sort().each { println it }
```

Comment:
❓ Wouldn't it make sense to use the Groovy [`relativePath`](https://docs.groovy-lang.org/latest/html/groovy-jdk/java/io/File.html#relativePath(java.io.File)) extension here?

Reply:
Yes, that would be better. Thank you 👍